src/segments/pulumi.go

```diff
@@ -0,0 +1,126 @@
+package segments
+
+import (
+	"encoding/json"
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+)
+
+type Pulumi struct {
+	props properties.Properties
+	env   platform.Environment
+
+	Stack string
+
+	Backend
+}
+
+type Backend struct {
+	URL  string
+	User string
+}
+
+type PulumiStackLSJson struct {
+	Name          string `json:"name"`
+	Current       bool   `json:"current"`
+	ResourceCount int    `json:"resourceCount"`
+	LastUpdate    string `json:"lastUpdate"`
+}
+
+type PulumiAboutJS struct {
+	Backend *PulumiBackendJS `json:"backend"`
+}
+
+type PulumiBackendJS struct {
+	URL  string `json:"url"`
+	User string `json:"user"`
+}
+
+func (p *Pulumi) Template() string {
+	return " {{if .User }}{{ .User }}@{{ end }}{{ if .URL }}{{ .URL }}{{ end }}{{if .Stack }} :: {{ .Stack }}{{ end }} "
+}
+
+func (p *Pulumi) Init(props properties.Properties, env platform.Environment) {
+	p.props = props
+	p.env = env
+}
+
+func (p *Pulumi) Enabled() bool {
+	if p.env.HasFiles("Pulumi.yaml") {
+		return p.getPulumiStackName() && p.getPulumiAbout()
```

Comment:
Do we need to make `fetch_about` and `fetch_stack` boolean properties? Now we always call this, but I can imagine not everyone wants to have that context.